stripe/resource.py

```diff
@@ -316,6 +316,22 @@ def retrieve(self, id, **params):
 
         return self.request('get', url, params)
 
+    def __iter__(self):
+        page = self
+        params = dict(self._retrieve_params)  # copy so not to mutate originals
+
+        while True:
+            last_id = None
+            for item in page.data:
+                last_id = item.get('id')
```

Comment:
We can't assume that all list objects will have an `id`. To be extra safe, we should use `.get('id', None)`